File: 0728_SelfDividing.py
#import math

class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        fout = []
        for num in range(left,right+1):
            isSelfDividing = True
            tenp = 1
            while tenp < num and isSelfDividing: #Innocent until proven guilty
                d = (num//tenp)%10
                if d != 0: #Sanity check
                    if num % d != 0:
                        isSelfDividing = False
                else:
                    isSelfDividing = False
                tenp *= 10
            if isSelfDividing:
                fout.append(num)
        return fout


        

File: 0728_SelfDividing.go
func selfDividingNumbers(left int, right int) []int {
    var tenp,d int
    var isSelfDividing bool
    fout := make([]int,0)
    for n:=left; n<=right; n++{
        isSelfDividing = true
        tenp = 1
        for tenp < n{
            d = (n/tenp)%10
            if d != 0 { //Sanity check
                if n%d != 0{
                    isSelfDividing = false
                    break
                }
            } else {
                isSelfDividing = false
                break
            }
            tenp *= 10
        }
        if isSelfDividing{
            fout = append(fout,n)
        }
    }
    return fout
}
